### PR TITLE
New package: wayfire-0.1

### PR DESCRIPTION
**Pull request by youngjinpark20**

### CODE REVIEW

**Comment by youngjinpark20** [comment]
https://github.com/void-linux/void-packages/issues/7521

**Comment by maxice8** [comment]
Fix the errors pointed out by Travis.

File: srcpkgs/wayfire/INSTALL.msg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Before running Wayfire, copy the default configuration file to your user:
+
+  cp /usr/share/wayfire/wayfire.ini.default ~/.config/wayfire.ini
+
+You also probably want to install elogind and dbus-elogind, then restart your computer
```
Comment (CameronNemo) [inline srcpkgs/wayfire/INSTALL.msg]: I don't think this needs dbus-elogind

File: srcpkgs/wayfire/template [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+makedepends="wlroots-devel glm cairo-devel wf-config libcap-devel"
+short_desc="3D wayland compositor"
+maintainer="Young Jin Park <youngjinpark20@gmail.com>"
+license="MIT"
+homepage="https://wayfire.org"
+distfiles="https://github.com/WayfireWM/wayfire/archive/v${version}.tar.gz"
+checksum=ce6ed0ba62296992f4a0c05b92d05b33911d1d192346611df845414ffd9a18b8
+
+post_install() {
+	vcopy wayfire.ini.default /usr/share/wayfire
```
Comment (CameronNemo) [inline srcpkgs/wayfire/template]: Can it not read /etc/wayfire/config like sway does?

File: srcpkgs/wayfire/template [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'wayfire'
+pkgname=wayfire
+version=0.1
+revision=1
+build_style=meson
+hostmakedepends="pkg-config cmake"
```
Comment (CameronNemo) [inline srcpkgs/wayfire/template]: Am I reading this right? It needs meson and cmake?

File: srcpkgs/wayfire/INSTALL.msg (hunk 1 above)
Comment (AlmightyFloppyFish) [reply srcpkgs/wayfire/INSTALL.msg]: Without elogind, it depends on systemd-login.

Comment (CameronNemo) [reply srcpkgs/wayfire/INSTALL.msg]: I said dbus-elogind. That is only related to specific dbus feature that as far as I can tell is only used by bluez.

File: srcpkgs/wayfire/template (hunk 2 above)
Comment (youngjinpark20) [reply srcpkgs/wayfire/template]: yeah it's honestly not too great in upstream

File: srcpkgs/wayfire/INSTALL.msg (hunk 1 above)
Comment (youngjinpark20) [reply srcpkgs/wayfire/INSTALL.msg]: removed dbus-elogind, I had assumed wrong :(

File: srcpkgs/wayfire/template (hunk 3 above)
Comment (youngjinpark20) [reply srcpkgs/wayfire/template]: you're right

File: srcpkgs/wayfire/template (hunk 2 above)
Comment (CameronNemo) [reply srcpkgs/wayfire/template]: Oh it is okay, just wanted to make sure. Also, I think using vinstall might be more robust than vcopy here.

Comment (youngjinpark20) [reply srcpkgs/wayfire/template]: you're really killing me here.... jk :smile: thanks for your feedback, fixed

**Comment by AlmightyFloppyFish** [comment]
It builds but I'm forced to run with `LD_PRELOAD=/usr/lib/libwlroots.so.0 wayfire` otherwise it fails with
`wayfire: symbol lookup error: wayfire: undefined symbol: wlr_pointer_gestures_v1_send_pinch_end` 

ldd shows that it defaults to /usr/local/lib/libwlroots.so.0

**Comment by CameronNemo** [comment]
Sounds like you have a manual installation of wlroots hanging around. Remove any wlroots related files in /usr/local and try again.

**Comment by maxice8** [comment]
You're using `vlicense` on a file that doesn't exist. vlicense is used to install a file on the license directory of a package.

You need to look into the source of the package and find the file that describe the license.

File: srcpkgs/wf-config/template [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+version=0.1
+revision=1
+build_style=meson
+hostmakedepends="pkg-config"
+makedepends="libevdev-devel wlroots-devel"
+short_desc="Wayfire library for managing configuration files"
+maintainer="Young Jin Park <youngjinpark20@gmail.com>"
+license="MIT"
+homepage="https://wayfire.org"
+distfiles="https://github.com/WayfireWM/wf-config/archive/v${version}.tar.gz"
```
Comment (maxice8) [inline srcpkgs/wf-config/template]: add this line to distfiles, below the current one (inside the double quotes that denote distfiles=.

https://raw.githubusercontent.com/WayfireWM/wf-config/master/LICENSE>LICENSE.txt

File: srcpkgs/wf-config/template [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+checksum=0dac4683013d5a0b143f0e211288c7f2f621740b9ac664359297ff530a7dcdc5
+
+post_install() {
+	vlicense LICENSE
```
Comment (maxice8) [inline srcpkgs/wf-config/template]: then adapt to vlicense ../LICENSE.txt

File: srcpkgs/wayfire/template (hunk 3 above)
Comment (ammen99) [reply srcpkgs/wayfire/template]: It shouldn't need cmake, are you sure it is needed? What error do you get without cmake?

Comment (maxice8) [reply srcpkgs/wayfire/template]: It was removed in the latest revision @ammen99

**Pull request closed without merging**